src/systemctl/systemctl-list-unit-files.c

```diff
@@ -143,27 +143,23 @@ int verb_list_unit_files(int argc, char *argv[], void *userdata) {
         bool fallback = false;
 
         if (install_client_side()) {
-                Hashmap *h;
+                _cleanup_(hashmap_freep) Hashmap *h = NULL;
                 UnitFileList *u;
                 unsigned n_units;
 
-                h = hashmap_new(&string_hash_ops);
+                h = hashmap_new(&unit_file_list_hash_ops);
                 if (!h)
                         return log_oom();
 
                 r = unit_file_get_list(arg_runtime_scope, arg_root, h, arg_states, strv_skip(argv, 1));
-                if (r < 0) {
-                        unit_file_list_free(h);
+                if (r < 0)
                         return log_error_errno(r, "Failed to get unit file list: %m");
-                }
 
                 n_units = hashmap_size(h);
 
                 units = new(UnitFileList, n_units ?: 1); /* avoid malloc(0) */
```

Comment:
`new()` uses `malloc_multiply()` which avoids calling `malloc(0)`. Please drop this work-around (in a separate commit ofc.)

Reply:
Done.